lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    Command.execute cmd, user_id
```

Comment:
Was referring to this idea you had above

```elixir
def execute(cmd, ctxt) do
    ctxt
    |> spawn_creature(Troll)
    |> notify_player(Message.info("You have sprung a trap!")
    |> set_timeout({:second, 60}, %ResetTrapEvent{ trap: cmd.target }) 
end
```
I saw that as building a context not doing anything. I may have interpreted that wrong, sorry. To me if we did that then the commands are very testable and maybe execute is the wrong word for that function.

With that in mind, where I was thinking it was just building a context (not doing anything) I was thinking that player process is ideal starting place. So the following is what I was thinking from your example above with the move command.
```elixir
# move command
def execute(cmd, ctx) do
  ctx
  |> move_player(cmd.exit)
  |> notify_player("You head #{cmd.exit}")
end

# player module
def handle_cast({:run_cmd, cmd}, _, state) do
  ctx = Game.Context.new
  |> add_character(state.character)
  |> add_room(state.room)
  
  ctx = Command.execute(cmd, ctx)

  Room.run ctx
end
```
Room.run can then use the context locate the room knows which player and can run the actions on the context in the room. I'm not sure a room cares about the commands, it just takes a context with actions and runs all the actions...

```elixir
# so after this
ctx = Game.Context.new
|> add_character(state.character)
|> add_room(state.room)

ctx = Command.execute(cmd, ctx)

# I imagine ctx looking like this
%Context{
  player: %Character{name: "that guy"},
  room: %Room{id: 123},
  actions: [
    move_player: {:exit, "north"},
    notify_player: {:message, "You head north"}
  ]
}
```
Actions could be funs to run instead of data structures etc, just an example but this is what is returned form Command.execute. Maybe it should be called Command.build(cmd, ctx) instead of execute... 

Anyway it's just what came to mind when you wrote that snippet of code out - then the rooms only care about context and it's how the player process talks to the game, because Room.run can dispatch the context to the correct room.